### PR TITLE
[ADF-5284] Restore original plugin functionality

### PR DESCRIPTION
**Pull request by pionnegru**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
[ADF-5284](https://alfresco.atlassian.net/browse/ADF-5284)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by eromano — APPROVED** [review]

**Review by DenysVuika — APPROVED** [review]

File: lib/extensions/src/lib/services/extension-loader.service.ts [hunk 1]
```diff
@@ -166,4 +166,12 @@ export class ExtensionLoaderService {
         }
         return action;
     }
+
+    private filterIgnoredExtensions(extensions: string[], config: ExtensionConfig): string[] {
```
Comment (popovicsandras) [inline lib/extensions/src/lib/services/extension-loader.service.ts:170]: Why passing down the whole config object? I assume the $ignoreReferenceList would be enough, this is the only thing what we use from this object, as I see.

File: lib/extensions/src/lib/services/extension-loader.service.ts [hunk 2]
```diff
@@ -42,6 +42,10 @@ export class ExtensionLoaderService {
                         config = JSON.parse(override);
                     }
 
+                    if (extensions && extensions.length) {
+                        config.$references.push(...this.filterIgnoredExtensions(extensions, config));
```
Comment (popovicsandras) [inline lib/extensions/src/lib/services/extension-loader.service.ts:46]: If we push, there is no way to override the the order. As I see, here we need to do is something similar to this:\\\\\\\\\

```js
// No $references defined in the app.extensions.json, so we can continue the dynamically registered ones
// coming from the extensions variable, paying attention to the ignored ones as well
if (!config.$references || !config.$references.length) {
	config.$references = this.filterIgnoredExtensions(extensions, config);
}
// User defined their own array of $references, we should proceed with that, paying attention
// to the ignored ones as well
else {
	config.$references = this.filterIgnoredExtensions(config.$references, config);
}
```

For this, the application (ACA/ADW) should not have anything in their $references by default, which means, we need to modify even ACA to dynamically register its plugins either.

**Review by popovicsandras — CHANGES_REQUESTED** [review]

File: lib/extensions/src/lib/services/extension-loader.service.ts (hunk 2 above)
Comment (pionnegru) [reply lib/extensions/src/lib/services/extension-loader.service.ts:46]: the  `push` is just to load the registered plugins.  Isn't the order set by the subsequent logic https://github.com/Alfresco/alfresco-ng2-components/pull/6348/files/a0fb40821839ddf565aa5f4b33e3cc885daac35f#diff-c6a1c872a637e729391870f0795194fb846e560b34090f242e98be038057977bL50-L54 ?

File: lib/extensions/src/lib/services/extension-loader.service.ts (hunk 1 above)
Comment (pionnegru) [reply lib/extensions/src/lib/services/extension-loader.service.ts:170]: ah, yes

**Review by popovicsandras — APPROVED** [review]

**Pull request merged**